### PR TITLE
Ruby: split standard library models into multiple files

### PR DESCRIPTION
**Pull request by hmac**
Split the classes modeling various standard library concepts into a
structured group of multiple files.

Things that are part of the core language live in libraries/core and
standard libraries (that aren't part of core) live in libraries/stdlib. In future we may want to add `libraries/gems` to hold third-party libraries. At the moment these live as separate files at the `libraries` level.

This mirrors the structure followed by the Ruby docs
(https://docs.ruby-lang.org/en/3.1/).

This PR is split up into multiple commits so it's easier to review. As a result, the intermediate commits will not build, but it should make reviewing a lot easier.

The last two commits are optional: they rename the `frameworks` directory to `libraries`, since I think that better describes its contents. This does churn many import paths, so I'm happy to drop this if you think it's too disruptive.

### CODE REVIEW

**Review by alexrford — COMMENTED** [review]
This is generally really nice and makes things much neater.

I'm conflicted on the `frameworks` -> `libraries` rename. It seems more accurate as to what we're actually modelling, but JS and Python generally bundle up both frameworks and libraries under the `frameworks` umbrella term. I also have some fears about breaking external queries and libraries that currently import QLLs that are moved or deleted in this PR.

On balance I'd personally prefer to keep these under `frameworks` for consistency with other languages.

For similar reasons of avoiding breaking external queries, we might want to keep a version of `frameworks/StandardLibrary.qll` around that re-exports public modules/classes/predicates of `Core.qll` and `Stdlib.qll` as `deprecated` versions. Ideally we would just be able to mark `StandardLibrary.qll` itself as `deprecated`, but I'm not aware of a simple way to do this.

**Comment by nickrolfe** [comment]
I guess this is going to clash with https://github.com/github/codeql/pull/7878, but I'm not sure how soon that PR is going to be in a mergeable state.

File: ruby/ql/lib/codeql/ruby/libraries/Core.qll
Comment (hmac) [reply ruby/ql/lib/codeql/ruby/libraries/Core.qll:13]: The reason I did this was to reduce churn in other files that previously imported `StandardLibrary.qll` and then used the classes directly. I can add a followup commit that explicitly imports the required modules in those files, and then change these imports back to `import core.Object; import core.Kernel`. Maybe that would be better as a followup, to avoid piling too many things in this PR? What do you think?

**Comment by hmac** [comment]
I added the `no-change-note-required` label because this PR doesn't change any queries, but perhaps it would be useful to add a change note about `StandardLibrary` being deprecated?

**Comment by asgerf** [comment]
> I guess this is going to clash with https://github.com/github/codeql/pull/7878, but I'm not sure how soon that PR is going to be in a mergeable state.

FWIW don't let the existence of that PR slow you down. I'm confident I can resolve conflicts efficiently on my end, and I'd be happy to help resolve conflicts in other PRs once mine is merged.

File: ruby/ql/lib/codeql/ruby/libraries/Core.qll
Comment (alexrford) [reply ruby/ql/lib/codeql/ruby/libraries/Core.qll:13]: Having this as a followup sounds good to me.

**Comment by nickrolfe** [comment]
If I understood correctly, sticking with the `frameworks` name will be the less disruptive option. I don't find myself strongly preferring either name over the other, so I guess I vote weakly for `frameworks`.

**Comment by hmac** [comment]
OK, let's go with frameworks.

**Review by alexrford — DISMISSED** [review]
The test output needs to be regenerated, otherwise LGTM.

**Comment by hmac** [comment]
Ugh that's so annoying! I thought this was ready to go...

**Review by alexrford — APPROVED** [review]
👍

**Pull request merged**
